### PR TITLE
Hide table and show message when a sample has 0 files

### LINKED ISSUE #152: Empty state for Sample > Files
Instead of showing an empty table when there are no files attached to a sample, hide the table and display a message that says there are no files attached to the sample.  Something like:

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## What does this PR do and why?
When a sample has no files, shows a message instead of rendering an empty table.
Fixes #152

## Screenshots or screen recordings
Sample with an attachment
image

Sample without any attachments
image

## How to set up and validate locally
1. pull branch, run, run tailwind to update ui
2. create a new project
3. create 2 samples
4. add an attachment to one of the samples
5. verify that the appropriate table or message is rendered depending on if the sample has attachments or not.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by JeffreyThiessen** [comment]
When there are attachments on a sample, and they are all deleted, the table stays on page empty. Refreshing shows the message as intended.

Should we be redrawing the cards on sample delete for this edge case?

**Comment by joshsadam** [comment]
Try a more centred layout.  A good reference you might want to take a look at is https://tailwindui.com/components/application-ui/feedback/empty-states#component-42930c785190e14c24b53ba822f4a92c

**Comment by joshsadam** [comment]
Can you update the screenshots for this PR?

**Comment by JeffreyThiessen** [comment]
> Can you update the screenshots for this PR?
Updated


> Can you add a test that ensures the empty state is displayed when there are no attachments?
I have added asserts to the existing test `'user with role >= Maintainer should be able to attach a file to a Sample'` which checks if the text/files are showing or not in the `#attachments` element before and after adding the file.

File: app/assets/stylesheets/application.tailwind.css [hunk 1]
```diff
@@ -465,3 +465,11 @@ body {
 div.field_with_errors > :is(select) {
   @apply text-red-900 placeholder-red-300 border-red-700;
 }
+
+.table-container table:not(:has(td)) {
```
[the remark this answers is not in the document]
Comment (ericenns) [reply app/assets/stylesheets/application.tailwind.css:469]: To solve this, just change the last added class to as follows:

```css
.table-container:not(has(td)) .empty_state_message {
  display: block;
}
```

and add the following:

```css
.table-container .empty_state_message {
  display: none;
}
```

[the remark this answers is not in the document]
Comment (JeffreyThiessen) [reply app/assets/stylesheets/application.tailwind.css:469]: Updated

[the remark this answers is not in the document]
Comment (joshsadam) [reply app/assets/stylesheets/application.tailwind.css:469]: Thanks

**Review by joshsadam — APPROVED** [review]
This looks good, thanks @JeffreyThiessen
